clients/roscpp/include/ros/common.h.in

```diff
@@ -48,6 +49,18 @@
 #define ROS_VERSION_GE(major1, minor1, patch1, major2, minor2, patch2) (ROS_VERSION_COMBINED(major1, minor1, patch1) >= ROS_VERSION_COMBINED(major2, minor2, patch2))
 #define ROS_VERSION_MINIMUM(major, minor, patch) ROS_VERSION_GE(ROS_VERSION_MAJOR, ROS_VERSION_MINOR, ROS_VERSION_PATCH, major, minor, patch)
 
+// check if we might need to include our own backported version boost::condition_variable
+// in order to use CLOCK_MONOTONIC for the condition variable and for the SteadyTimer
+#cmakedefine ROSCPP_USE_BACKPORTED_BOOST_CONDITION_VARIABLE_IMPLEMENTATION
+#ifdef ROSCPP_USE_BACKPORTED_BOOST_CONDITION_VARIABLE_IMPLEMENTATION
+  // use backported version of boost condition variable, see https://svn.boost.org/trac/boost/ticket/6377
+  #define ROSCPP_BOOST_CONDITION_VARIABLE_HEADER "boost_161_condition_variable.h"
+  #define ROSCPP_BOOST_CONDITION_VARIABLE boost_161::condition_variable
```

Comment:
Wouldn't it be better/cleaner/more convenient to use a typedef here?

Reply:
> Wouldn't it be better/cleaner/more convenient to use a typedef here?

Indeed: https://github.com/ros/ros_comm/pull/1651/commits/05f86d950ae137ca4ef521a20f183cd5d76e1fd4